app/components/alerts/AlertsList.js

```diff
@@ -5,46 +5,82 @@ import { RefreshControl, FlatList, ScrollView, View, Image, Text } from 'react-n
 import { ListItem } from 'react-native-elements';
 import AlertsStyles from '../../styles/AlertStyles';
 import emptyImage from '../../images/alerts.png';
-import { getAlertsList } from '../../actions';
+import { getAlertsList, updateUserLastAlertSeen } from '../../actions';
 
 class AlertsList extends React.Component {
-  constructor(props) {
-    super(props);
+	constructor(props) {
+		super(props);
 
-    this.state = {
+		props.currentAlerts.sort(function(a,b){
+			return (b.sentDate == null)-(a.sentDate == null) || b.sentDate - a.sentDate;
+		}).reverse();
+
+		this.state = {
+			userId: props.userId,
+			lastAlertSeen: props.lastAlertSeen,
 			getAlertsList: props.getAlertsList,
+			updateUserLastAlertSeen: props.updateUserLastAlertSeen,
 			currentAlerts: props.currentAlerts,
 			isRefreshing: false,
-    };
+		};
 
-    this.renderListItem = this.renderListItem.bind(this);
-    this.onRefresh = this.onRefresh.bind(this);
-  }
+    this.state.getAlertsList(); //ensure newest alerts shown w/o refreshing
+
+		this.renderListItem = this.renderListItem.bind(this);
+		this.onRefresh = this.onRefresh.bind(this);
+	}
 
-  componentWillReceiveProps(nextProps) {
+	componentWillReceiveProps(nextProps) {
 		// Avoiding refresh if possible
 		if (nextProps.currentAlerts !== this.state.currentAlerts) {
```

Comment:
I recommend using `arrayOfObjectEquals` from `utils/arrays.js` here because comparing JS objects with equality sign does a loose equality and does not check if the children are the same.  Check [this](http://adripofjavascript.com/blog/drips/object-equality-in-javascript.html) out.